packages/react-vanilla-components/src/components/FileUpload.tsx

```diff
@@ -52,33 +67,75 @@ const FileUpload = (props: PROPS) => {
     return localFiles;
   }, []);
 
-  const handleNewFileUpload = useCallback((e: { target: { files: FileList | null; }; }) => {
-    const { files: newFiles } = e.target;
-    if (newFiles?.length) {
-      const updatedFiles = files.concat(fileListToArray(newFiles));
+  const handleDragOver = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    const tgt = event.target as Element;
+    console.log(tgt);
+    tgt.classList.add('cmp-adaptiveform-fileinput__dragarea--active');
+  };
+
+  const handleDragLeave = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    const tgt = event.target as Element;
+    tgt.classList.remove('cmp-adaptiveform-fileinput__dragarea--active');
```

Comment:
add/remove class based on the state. Don’t use the js selector in react.

Reply:
Implemented!